core/app/services/spree/checkout/update.rb

```diff
@@ -4,10 +4,32 @@ class Update
       prepend Spree::ServiceModule::Base
 
       def call(order:, params:, permitted_attributes:, request_env:)
+        params = transfer_addresses_params(params) if order_address_present?(params)
         return success(order) if order.update_from_params(params, permitted_attributes, request_env)
 
         failure(order)
       end
+
+      private
+
+      def order_address_present?(params)
+        return false if params[:order].empty?
+
+        (params[:order][:bill_address_attributes] || params[:order][:ship_address_attributes]).present?
+      end
+
+      def transfer_addresses_params(params)
+        bill_addr_country_id = Country.country_id_by_iso(params[:order][:bill_address_attributes].fetch(:country_iso))
```

Comment:
@przemosk we should only run those check if `country_iso` are present and `country_id` is blank/null

Also - let's add some checks for this method ok?